File: src/test/java/com/ragin/bdd/cucumbertests/hooks/ParameterizedCustomScenarioContextMatcher.java
package com.ragin.bdd.cucumbertests.hooks;

import com.ragin.bdd.cucumber.core.ScenarioStateContext;
import com.ragin.bdd.cucumber.matcher.BddCucumberJsonMatcher;
import net.javacrumbs.jsonunit.core.ParametrizedMatcher;
import org.hamcrest.BaseMatcher;
import org.hamcrest.Description;

public class ParameterizedCustomScenarioContextMatcher extends BaseMatcher<Object> implements BddCucumberJsonMatcher, ParametrizedMatcher {
    private String jsonParameter;

    @Override
    public String matcherName() {
        return "isInContextAvailable";
    }

    @Override
    public Class<? extends BaseMatcher<?>> matcherClass() {
        return this.getClass();
    }

    @Override
    public void setParameter(String parameter) {
        this.jsonParameter = parameter;
    }

    @Override
    public boolean matches(final Object actual) {
        final String actualParameter = String.valueOf(actual);
        final String contextParameter = ScenarioStateContext
                .current()
                .getScenarioContextMap()
                .get(jsonParameter);
        return actualParameter.equals(contextParameter);
    }

    @Override
    public void describeTo(Description description) {
    }
}

File: src/test/kotlin/com/ragin/bdd/cucumbertests/hooks/ParameterizedCustomScenarioContextMatcher.kt
package com.ragin.bdd.cucumbertests.hooks

import com.ragin.bdd.cucumber.core.ScenarioStateContext.current
import com.ragin.bdd.cucumber.matcher.BddCucumberJsonMatcher
import net.javacrumbs.jsonunit.core.ParametrizedMatcher
import org.hamcrest.BaseMatcher
import org.hamcrest.Description

class ParameterizedCustomScenarioContextMatcher : BaseMatcher<Any>(), BddCucumberJsonMatcher, ParametrizedMatcher {
    private var jsonParameter: String? = null

    override fun matcherName(): String {
        return "isInContextAvailable"
    }

    override fun matcherClass(): Class<out BaseMatcher<*>?> {
        return this.javaClass
    }

    override fun setParameter(parameter: String) {
        this.jsonParameter = parameter
    }

    override fun matches(actual: Any): Boolean {
        val actualParameter = actual.toString()
        val contextParameter = current()
            .scenarioContextMap[jsonParameter]
        return actualParameter == contextParameter
    }

    override fun describeTo(description: Description) = Unit
}
